tests/tests/acceptance/ReplacementCest.php

```diff
@@ -255,4 +257,98 @@ public function createAnExpectationWithoutRegexMatchGroups(AcceptanceTester $I)
+    {
+        $expectation = PhiremockClient::on(
+            A::getRequest()->andUrl(Is::matching('/&test=(\d+)/'))
+        )->then(
+            Respond::withStatusCode(200)
+                ->andHeader('X-Header', 'test=${url.1}')
+        );
+        $this->phiremock->createExpectation($expectation);
+        
+        $I->sendGET('/potato', ['param1' => 123, 'test' => 456]);
+        $I->seeResponseCodeIs('200');
+        $I->canSeeHttpHeader('X-Header', 'test=456');
+    }
+    
+    public function createAnExpectationWithRegexReplacementInHeaderFromBody(AcceptanceTester $I)
+    {
+        $expectation = PhiremockClient::on(
+            A::postRequest()->andUrl(Is::equalTo('/potato'))
+                ->andBody(Is::matching('/a tomato (\d+)/'))
+        )->then(
+            Respond::withStatusCode(200)
+                ->andHeader('X-Header', '${body.1}')
+        );
+        $this->phiremock->createExpectation($expectation);
+    
+        $I->sendPOST('/potato', 'this is a tomato 3kg it weights');
+        $I->seeResponseCodeIs('200');
+        $I->canSeeHttpHeader('X-Header', '3');
+    }
+    
+    public function createAnExpectationWithRegexReplacementInHeaderFromBodyAndUrl(AcceptanceTester $I)
+    {
+        $expectation = PhiremockClient::on(
+            A::postRequest()->andUrl(Is::matching('~^/potato/(\d+)$~'))
+                ->andBody(Is::matching('/^this is a tomato (\d+)kg it weights$/'))
+        )->then(
+            Respond::withStatusCode(200)
+                ->andBody('the numbers are ${url.1} and ${body.1}')
+                ->andHeader('X-Header', 'url=${url.1} body=${body.1}')
+        );
+        $this->phiremock->createExpectation($expectation);
+    
+        $I->sendPOST('/potato/456', 'this is a tomato 3kg it weights');
+        $I->seeResponseCodeIs('200');
+        $I->seeResponseEquals('the numbers are 456 and 3');
+        $I->canSeeHttpHeader('X-Header', 'url=456 body=3');
+    }
+    
+    public function createAnExpectationWithRegexReplacementInHeaderAsGroupExpression(AcceptanceTester $I)
+    {
+        $expectation = PhiremockClient::on(
+            A::postRequest()->andUrl(Is::matching('~^/potato/(\d+)$~'))
+                ->andBody(Is::matching('/^this is a tomato (\d+)kg it weights$/'))
+        )->then(
+            Respond::withStatusCode(200)
+                ->andBody('the numbers are ${url.1.1} and ${body.1.1}')
+                ->andHeader('X-Header', 'url=${url.1.1} body=${body.1.1}')
+        );
+        $this->phiremock->createExpectation($expectation);
+        
+        $I->sendPOST('/potato/456', 'this is a tomato 3kg it weights');
+        $I->seeResponseCodeIs('200');
+        $I->seeResponseEquals('the numbers are 456 and 3');
+        $I->canSeeHttpHeader('X-Header', 'url=456 body=3');
+    }
+    
+    public function createAnExpectationWithRegexReplacementInHeaderWithMultipleMatchGroups(AcceptanceTester $I)
```

Comment:
I tried to mirror the majority the tests for URL and Body replacements. I can add more if desired.

Reply:
They look awesome. Thanks